packages/checkout/sdk/src/smartCheckout/sell/sell.ts

```diff
@@ -130,18 +131,19 @@ export const sell = async (
     orderbook = instance.createOrderbookInstance(config);
     const { seaportContractAddress } = orderbook.config();
     spenderAddress = seaportContractAddress;
-    const sellItem = sellToken.type === ItemType.ERC721
+
+    const sellItem = 'type' in sellToken && sellToken.type === ItemType.ERC1155
       ? {
         type: sellToken.type,
         contractAddress: sellToken.collectionAddress,
         tokenId: sellToken.id,
+        amount: sellToken.amount,
       }
       : {
-        type: sellToken.type,
+        type: ItemType.ERC721,
         contractAddress: sellToken.collectionAddress,
         tokenId: sellToken.id,
-        amount: sellToken.amount,
-      };
+      } as OrderbookERC721Item;
```

Comment:
Wonder why is necessary to force the type here? Wouldn't `type: ItemType.ERC721` get typescript to detect this as an `ERC721` already?

Reply:
Yeh I think due to the ternary it thought otherwise `amount` as undefined, which strictly isnt on the orderbook ERC721 item type. I think its cleaner to not cast here and instead strictly type the declared const which I have now done